src/components/ButtonOption/index.js

```diff
@@ -0,0 +1,59 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import { ButtonGroupPickerContext } from '../ButtonGroupPicker/context';
+import HiddenElement from '../Structural/hiddenElement';
+import StyledLabel from './styled/label';
+import StyledText from './styled/text';
+import isOptionSelected from './helpers/isOptionSelected';
+import { uniqueId } from '../../libs/utils';
+
+export default function ButtonOption(props) {
+    const inputId = uniqueId('button-option');
```

Comment:
this will cause the inputId change in each rerender, that is why we normally use this in the constructor
for functional components using hooks we created a custom hook that get called only onces
`import { useUniqueIdentifier } from '../../libs/hooks';`
use this instead